src/main/groovy/edu/oregonstate/mist/invencycle/db/BikeDAO.groovy

```diff
@@ -0,0 +1,79 @@
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
+import org.skife.jdbi.v2.sqlobject.SqlQuery
+import org.skife.jdbi.v2.sqlobject.SqlUpdate
+import org.skife.jdbi.v2.sqlobject.Bind
+
+@RegisterMapper(bikemapper)
+public interface BikeDAO {
+
+    @SqlQuery("""
+        SELECT
+          BIKE.BIKE_ID,
+          BIKE.BIKE_MAKE,
+          BIKE.BIKE_MODEL,
+          BIKE.BIKE_YEAR,
+          BIKE_TYPE.BIKE_TYPE_NAME,
+          FRAME_SIZE.FRAME_SIZE_NAME,
+          FRAME_SIZE.FRAME_SIZE_CM,
+          BIKE.BIKE_MSRP,
+          D1.DERAILUER_MAKE AS DERAILUER_MAKE_FRONT,
+          D1.DERAILUER_MODEL AS DERAILUER_MODEL_FRONT,
+          D1.DERAILUER_SPEEDS AS DERAILUER_SPEEDS_FRONT,
+          D2.DERAILUER_MAKE AS DERAILUER_MAKE_REAR,
+          D2.DERAILUER_MODEL AS DERAILUER_MODEL_REAR,
+          D2.DERAILUER_SPEEDS AS DERAILUER_SPEEDS_REAR,
+          F.SUSPENSION_MAKE AS FORK_MAKE,
+          F.SUSPENSION_MODEL AS FORK_MODEL,
+          F.SUSPENSION_TRAVEL AS FORK_TRAVEL,
+          S.SUSPENSION_MAKE AS SHOCK_MAKE,
+          S.SUSPENSION_MODEL AS SHOCK_MODEL,
+          S.SUSPENSION_TRAVEL AS SHOCK_TRAVEL,
+          WF.WHEEL_SIZE AS WHEEL_SIZE_FRONT,
+          WF.WHEEL_RIM_MAKE AS RIM_MAKE_FRONT,
+          WF.WHEEL_RIM_MODEL AS RIM_MODEL_FRONT,
+          WF.WHEEL_HUB_MAKE AS HUB_MAKE_FRONT,
+          WF.WHEEL_HUB_MODEL AS HUB_MODEL_FRONT,
+          WR.WHEEL_SIZE AS WHEEL_SIZE_REAR,
+          WR.WHEEL_RIM_MAKE AS RIM_MAKE_REAR,
+          WR.WHEEL_RIM_MODEL AS RIM_MODEL_REAR,
+          WR.WHEEL_HUB_MAKE AS HUB_MAKE_REAR,
+          WR.WHEEL_HUB_MODEL AS HUB_MODEL_REAR,
+          BF.BRAKE_MAKE AS BRAKE_MAKE_FRONT,
+          BF.BRAKE_MODEL AS BRAKE_MODEL_FRONT,
+          BF.BRAKE_ROTOR_SIZE AS BRAKE_ROTOR_SIZE_FRONT,
+          BR.BRAKE_MAKE AS BRAKE_MAKE_REAR,
+          BR.BRAKE_MODEL AS BRAKE_MODEL_REAR,
+          BR.BRAKE_ROTOR_SIZE AS BRAKE_ROTOR_SIZE_REAR,
+          BIKE.BIKE_CREATE_DATE,
+          BIKE.BIKE_DELETE_DATE
+
+        FROM BIKE
+
+        LEFT JOIN BIKE_TYPE
+          ON BIKE.BIKE_BIKE_TYPE_ID = BIKE_TYPE.BIKE_TYPE_ID
+        LEFT JOIN FRAME_SIZE
+          ON BIKE.BIKE_FRAME_SIZE_ID = FRAME_SIZE.FRAME_SIZE_ID
+        LEFT JOIN DERAILUER D1
+          ON D1.DERAILUER_ID = BIKE.BIKE_DERAILUER_FRONT_ID
+        LEFT JOIN DERAILUER D2
+          ON D2.DERAILUER_ID = BIKE.BIKE_DERAILUER_REAR_ID
+        LEFT JOIN SUSPENSION F
+          ON F.SUSPENSION_ID = BIKE.BIKE_FORK_ID
+        LEFT JOIN SUSPENSION S
+          ON S.SUSPENSION_ID = BIKE.BIKE_SHOCK_ID
+        LEFT JOIN WHEEL WF
+          ON WF.WHEEL_ID = BIKE.BIKE_WHEEL_FRONT_ID
+        LEFT JOIN WHEEL WR
+          ON WR.WHEEL_ID = BIKE.BIKE_WHEEL_REAR_ID
+        LEFT JOIN BRAKE BF
+          ON BF.BRAKE_ID = BIKE.BIKE_BRAKE_FRONT_ID
+        LEFT JOIN BRAKE BR
+          ON BR.BRAKE_ID = BIKE.BIKE_BRAKE_REAR_ID
+        WHERE BIKE.BIKE_ID = :id
+        """)
+    Bike getById(@Bind("id") Integer id)
+}
```

Comment:
You need an override close at in this DAO. See: https://github.com/osu-mist/locations-api/blob/develop/src/main/groovy/edu/oregonstate/mist/locations/db/CampusMapLocationDAO.groovy